src/common/mfu_flist_copy.c

```diff
@@ -1554,46 +1574,80 @@ static int mfu_is_eof(const char* file, int fd)
 static int mfu_copy_file_normal(
     const char* src,
     const char* dest,
-    const int in_fd,
-    const int out_fd,
     uint64_t offset,
     uint64_t length,
     uint64_t file_size,
-    mfu_copy_opts_t* mfu_copy_opts)
+    mfu_copy_opts_t* mfu_copy_opts,
+    mfu_file_t* mfu_src_file,
+    mfu_file_t* mfu_dst_file)
 {
-    /* hint that we'll read from file sequentially */
-//    posix_fadvise(in_fd, offset, chunk_size, POSIX_FADV_SEQUENTIAL);
-
+    /* There is no equivalent "lseek" in daos, it is part of read, so skip that
+     * for a DAOS path */
     /* seek to offset in source file */
-    if(mfu_lseek(src, in_fd, offset, SEEK_SET) == (off_t)-1) {
-        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in source path `%s' (errno=%d %s)",
-            src, errno, strerror(errno));
+    if(mfu_file_lseek(src, mfu_src_file, offset, SEEK_SET) == (off_t)-1) {
+        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in source path `%s' (errno=%d %s)", src, errno, strerror(errno));
         return -1;
     }
 
     /* seek to offset in destination file */
-    if(mfu_lseek(dest, out_fd, offset, SEEK_SET) == (off_t)-1) {
-        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in destination path `%s' (errno=%d %s)",
-            dest, errno, strerror(errno));
+    if (mfu_file_lseek(dest, mfu_dst_file, offset, SEEK_SET) == (off_t)-1) {
+        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in destination path `%s' (errno=%d %s)", dest, errno, strerror(errno));
         return -1;
     }
 
-    /* get buffer */
-    size_t buf_size = mfu_copy_opts->block_size;
-    void* buf = mfu_copy_opts->block_buf1;
+    /* declare daos vars */
+#ifdef DAOS_SUPPORT
+    daos_size_t		got_size;
+    d_sg_list_t		sgl;
+    d_iov_t		iov;
+#endif
+
+    /* declare buffers */
+    size_t buf_size;
+    void* buf;
+    void* newbuf;
+    
+    /* set buffer and buffer size */
+    buf_size = mfu_copy_opts->block_size;
+    buf = mfu_copy_opts->block_buf1;
+
+    /* set daos vars and buffers */
+#ifdef DAOS_SUPPORT
+    mfu_src_file->offset = (daos_off_t) offset;
+    mfu_dst_file->offset = (daos_off_t) offset;
+    sgl.sg_nr = 1;
+    D_ALLOC(buf, buf_size);
+    D_ASSERT(buf != NULL);
+    d_iov_set(&iov, buf, buf_size);
+    sgl.sg_iovs = &iov;
+    sgl.sg_nr_out = 1;
+    mfu_src_file->sgl = &sgl;
+    mfu_dst_file->sgl = &sgl;
+#endif
```

Comment:
It looks like this code will execute regardless of whether the user is dealing with a DAOS object or not.  Let's only execute this if DAOS is being used.

Can you describe what the ```d_iov_set``` function does?

Reply:
the iovec is actually defined in Cart/Gurt. It is setting the iov_buf and iov_len to the allocated buffer. 
```C
static inline void
d_iov_set(d_iov_t *iov, void *buf, size_t size)
{
        iov->iov_buf = buf;
        iov->iov_len = iov->iov_buf_len = size;
}

```